packages/circuit-ui/components/NotificationInline/NotificationInline.tsx

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HTMLAttributes, RefObject, useEffect, useRef, useState } from 'react';
+import { css } from '@emotion/react';
+import { Alert, Confirm, Info, Notify } from '@sumup/icons';
+
+import styled, { StyleProps } from '../../styles/styled';
+import { useAnimation } from '../../hooks/useAnimation';
+import Body from '../Body';
+import { TrackingProps } from '../../hooks/useClickEvent';
+import CloseButton from '../CloseButton';
+import { hideVisually } from '../../styles/style-mixins';
+import Button, { ButtonProps } from '../Button';
+import { ClickEvent } from '../../types/events';
+
+const TRANSITION_DURATION = 200;
+const DEFAULT_HEIGHT = 'auto';
+
+type Variant = 'info' | 'confirm' | 'notify' | 'alert';
+
+type Action = ButtonProps;
+
+type CloseProps =
+  | {
+      /**
+       * Renders a close button in the top right corner and calls the provided function
+       * when the button is clicked.
+       */
+      onClose: (event: ClickEvent) => void;
+      /**
+       * Text label for the close button for screen readers.
+       * Important for accessibility.
+       */
+      closeButtonLabel: string;
+    }
+  | { onClose?: never; closeButtonLabel?: never };
+
+export type BaseProps = HTMLAttributes<HTMLDivElement> & {
```

Comment:
Can we share some of these props between the toast and this component? There should be few differences